frontend/src/Components/Karma.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useEffect, useState} from 'react';
+import './KarmaCalculator.scss';
+export function Karma() {
+    const [allPostsValue, setP] = useState(0); // sum of votes for all posts
+    const [allCommentsValue, setC] = useState(0); // sum of votes for all comments
+    const [profileVotesCount, setUCount] = useState(0); //count of votes in profile
+    const [profileVotingResult, setU] = useState(0); // sum of votes in profile
+    const [punishment, setPunishment] = useState(0); // penalty set by moderator
+
+    const [contentRating, setContentRating] = useState(0); // content quality ratio
+    const [userRating, setUserRating] = useState(0); //user reputation ratio
+    const [rawKarma, setRawKarma] = useState(0); // karma without punishment
+    const [karma, setKarma] = useState(0);// karma with punishment
+
+    const positiveCommentsDivisor = 10; // positive comments are 10x cheaper than posts
+    const negativeCommentsDivisor = 2; // negative comments are only 2x cheaper than posts
+    //TODO extract the rest of coefficients to constants and comment them
+
+    useEffect(() => {
+        setU( clamp(profileVotingResult, -profileVotesCount, profileVotesCount));
+    },[profileVotesCount]);
+
+    useEffect(() => {
```

Comment:
Fyi, you don't need state and effects for every part of the formula.
State is necessary only for the slider values (that user can change).
Every drived value can be calculated using statements and constants (without the effects).

Reply:
fixed

Reply:
Looks good to me.